ext/RMagick/extconf.rb

```diff
@@ -14,11 +14,22 @@ class Extconf
     attr_reader :headers
 
     def initialize
+      setup_paths_for_homebrew
       configure_compile_options
       assert_can_compile!
       configure_headers
     end
 
+    def setup_paths_for_homebrew
+      return unless find_executable('brew')
+
+      brew_pkg_config_path = "#{`brew --prefix imagemagick@6`.strip}/lib/pkgconfig"
+      pkgconfig_paths = ENV['PKG_CONFIG_PATH'].to_s.split(':')
+      if File.exist?(brew_pkg_config_path) && !pkgconfig_paths.include?(brew_pkg_config_path)
+        ENV['PKG_CONFIG_PATH'] = [ENV['PKG_CONFIG_PATH'], brew_pkg_config_path].compact.join(':')
```

Comment:
Do we need the `compact` here? I think we can do:

```rb
ENV['PKG_CONFIG_PATH'] = "#{ENV['PKG_CONFIG_PATH']}:#{brew_pkg_config_path}"
```

Reply:
If user does not set `PKG_CONFIG_PATH`, `ENV['PKG_CONFIG_PATH']` will return `nil`.

```
>> ENV['PKG_CONFIG_PATH']
=> nil
```

Then `"#{ENV['PKG_CONFIG_PATH']}:#{brew_pkg_config_path}"` will generate `:/usr/local/opt/imagemagick@6` path and `pkg-config` does not accept the path in my environment :(

Reply:
Ah, I see.